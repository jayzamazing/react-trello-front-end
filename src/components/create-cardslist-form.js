import React from 'react';
import {reduxForm, Field, focus} from 'redux-form';
import Input from './input';
import {required, nonEmpty, length, isTrimmed} from '../validators';

export function CreateCardslistForm(props) {
  return (
    <form className="create-cardslist-area cardslist-tile"
      onSubmit={
        props.handleSubmit(values => {
          props.onSubmit(values.cardslistTitle, props.boardId, props.board);
      })}>
        <Field
          component={Input}
          type="text"
          name="cardslistTitle"
          validate={[required, nonEmpty, isTrimmed]}
          placeholder="Add a list"
          labelclass="remove"
          inputClass="updateCardslist"
        />
        <button className="create-cardslist-btn btn btn-success"
          type="submit"
          disabled={props.pristine || props.submitting}>
          Save
        </button>
        <button type="button" className="btn btn-default close-cardslist-btn"
          aria-label="close button" onClick={() => props.close()}>
            <span className="glyphicon glyphicon-remove"
              aria-hidden="true"></span>
        </button>
      </form>
  );
}
export default reduxForm({
    form: 'createcardslist',
    onSubmitFail: (errors, dispatch) =>
        dispatch(focus('createcardslist', Object.keys(errors)[0]))
})(CreateCardslistForm);
