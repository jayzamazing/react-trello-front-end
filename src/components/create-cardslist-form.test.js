'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {CreateCardslistForm} from './create-cardslist-form';

describe('create-cardslist-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const handleSubmit = jest.fn();
    shallow(<CreateCardslistForm handleSubmit={handleSubmit}/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<CreateCardslistForm handleSubmit={handleSubmit}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on create button and check dispatches
  it('add board click', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<CreateCardslistForm handleSubmit={handleSubmit}/>);
    wrapper.find('[name="cardslistTitle"]').at(0).simulate('change', { target: { value: 'testing' } });
    wrapper.find('.create-cardslist-btn').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
  //click on create button and check dispatches
  it('close button', () => {
    const handleSubmit = jest.fn();
    const closeModal = jest.fn();
    const wrapper = shallow(<CreateCardslistForm close={closeModal} handleSubmit={handleSubmit}/>);
    wrapper.find('[aria-label="close button"]').simulate('click');
    expect(closeModal).toHaveBeenCalled();
  });
});
