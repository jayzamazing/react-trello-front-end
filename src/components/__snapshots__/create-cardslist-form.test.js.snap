// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-cardslist-form snapshot 1`] = `
<form
  className="create-cardslist-area cardslist-tile"
>
  <Field
    component={[Function]}
    inputClass="updateCardslist-input"
    labelclass="remove"
    name="cardslistTitle"
    placeholder="Add a list"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <button
    className="create-cardslist-btn btn btn-success"
    type="submit"
  >
    Save
  </button>
  <button
    aria-label="close button"
    className="btn btn-default close-cardslist-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-remove"
    />
  </button>
</form>
`;
